src/k8s/pkg/k8sd/setup/kube_proxy_test.go

```diff
@@ -0,0 +1,72 @@
+package setup_test
+
+import (
+	"context"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/canonical/k8s/pkg/k8sd/setup"
+	"github.com/canonical/k8s/pkg/snap/mock"
+	snaputil "github.com/canonical/k8s/pkg/snap/util"
+	. "github.com/onsi/gomega"
+)
+
+func TestKubeProxy(t *testing.T) {
+	g := NewWithT(t)
+
+	dir := t.TempDir()
+
+	s := &mock.Snap{
+		Mock: mock.Mock{
+			KubernetesConfigDir: path.Join(dir, "kubernetes"),
```

Comment:
Also set the `ServiceArgumentsDir` here https://github.com/canonical/k8s-snap/blob/main/src/k8s/pkg/k8sd/setup/containerd_test.go#L32